openmetadata-ui/src/main/resources/ui/src/components/common/CustomPropertyTable/PropertyInput.test.tsx

```diff
@@ -11,63 +11,67 @@
  *  limitations under the License.
  */
 
-import { fireEvent, render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import React from 'react';
-import { PropertyInput } from './PropertyInput';
+import { PropertyInput, PropertyInputProps } from './PropertyInput';
 
 const onCancel = jest.fn();
 const onSave = jest.fn();
 
-const mockProp = {
+const mockProp: PropertyInputProps = {
   value: 'yValue',
-  type: 'string',
+  type: 'text',
   propertyName: 'yNumber',
   onCancel,
   onSave,
 };
 
+jest.mock('components/InlineEdit/InlineEdit.component', () => {
+  return jest.fn().mockImplementation(({ children, onCancel, onSave }) => (
+    <div data-testid="inline-edit">
+      {children}
+      <button data-testid="cancel" onClick={onCancel}>
+        cancel
+      </button>
+      <button data-testid="save" onClick={onSave}>
+        save
+      </button>
+    </div>
+  ));
```

Comment:
We are rewriting the whole component here and testing it again which is not required. 
We can simply mock it and check whether it's being rendered or not that's it

Reply:
For cancel we can avoid but with the save button, we are testing onSave function that's why we need the save button in mock component

Reply:
For future references:
Unit tests only ment to tests components on isolation.
No end to end tests or user flow testing can covered in unit tests 😅

Reply:
sure, thanks you 👍